src/endpoint/endpoint.js

```diff
@@ -158,16 +158,15 @@ async function main(options = {}) {
             https_server.setSecureContext(updated_ssl_options);
             https_server_sts.setSecureContext(updated_ssl_options);
         });
-        const allow_http = await endpoint_utils.get_nsfs_system_property('allow_http', options.nsfs_config_root);
-        if (options.nsfs_config_root && !allow_http) {
-            dbg.log0('HTTP is not allowed for NSFS.');
-        } else {
+        if (await is_http_allowed(options.nsfs_config_root)) {
```

Comment:
is_https_allowed() can be normal (sync) function ? without await ?

Reply:
no @v-harihar `get_nsfs_system_property()` is a async function